kaggle/api/kaggle_api_extended.py

```diff
@@ -86,98 +86,248 @@ class KaggleApi(KaggleApi):
         reload(sys)
         sys.setdefaultencoding('latin1')
 
+## Authentication
+
     def authenticate(self):
-        try:
-            configuration = Configuration()
-            if os.name != 'nt':
-                permissions = os.stat(self.config).st_mode
-                if (permissions & 4) or (permissions & 32):
-                    print('Warning: Your Kaggle API key is readable by other '
-                          'users on this system! To fix this, you can run '
-                          '\'chmod 600 {}\''.format(self.config))
-            with open(self.config, 'r') as f:
-                config_data = json.load(f)
+        '''authenticate the user with the Kaggle API. This method will generate
+           a configuration, first checking the environment for credential
+           variables, and falling back to looking for the .kaggle/kaggle.json
+           configuration file.
+        '''
+        config_data = {}
+
+        # Step 1: read in configuration file, if it exists
+        if os.path.exists(self.config):
+            config_data = self._authenticate_config(config_data)
+
+        # Step 2:, get username/password from environment
+        config_data = self._authenticate_environment(config_data)
+
+
+        # Step 3: load into configuration!
+        self._load_config(config_data)
+
+
+    def _authenticate_environment(self, config_data=None, quiet=False):
```

Comment:
I would rename to something like `_read_config_env_variables`.  Naming made me do a few double takes.  If you have other ideas, I'm all ears, because I'm not super fond of that suggestion either.

Naming is no fun :(

Reply:
How about `authenticate_from_environment` ?

Reply:
or `auth_from_environment` or `authenticate_from_env`

Reply:
Let's consolidate the discussion down in the `_authenticate_config` method area

Reply:
okay!